### ISSUE TITLE
Evaluate variables in prop types

### ISSUE THREAD

**Issue by siddharthkp** [auth0/cosmos#253]
TLDR: To make icon names a dropdown instead of string in the docs, I need to invest a day in some configuration crap. Need help to decide priority.

--

Long story:

This one took me by surprise!

Working on https://github.com/auth0/cosmos/issues/231, I landed on docs for [`Icons`](https://auth0-cosmos.now.sh/docs/Icon), this is a beautiful component. 

@nkohari Sorry if you asked about this earlier and I missed the conversation.

Here are the prop types for Icon:

```js
Icon.propTypes = {
  /** Icon name */
  name: PropTypes.string.isRequired,
  /** Icon size */
  size: PropTypes.number,
  /** Icon foreground color */
  color: PropTypes.string
}
```

Now that we have `enum` support in the docs, I wanted `name` to be a dropdown from the list of icons we have.

So I changed the prop-types definitions

```diff
+ const iconNames =  Object.keys(icons)

Icon.propTypes = {
  /** Icon name */
-  name: PropTypes.string.isRequired,
+  name: PropTypes.oneOf(iconNames).isRequired,
  /** Icon size */
  size: PropTypes.number,
  /** Icon foreground color */
  color: PropTypes.string
}
```

Turns out, `react-docgen` does not evaluate variables inside propTypes, which makes the prop table useless

```js
{
  name: "enum",
  value: "iconNames"
}
```

A nice and clean way to fix this is to cause evaluation the value of `iconNames`. [`preval`](https://github.com/kentcdodds/babel-plugin-preval) is a nice way of doing it, but it requires modifying the babel config. As we use `create-react-app`, we don't have access to that configuration 🙃

We can eject (sad) or use something like [react-app-rewired](https://github.com/timarney/react-app-rewired) to gain control of the config. 

OR we can upgrade to `react-scripts-next` (experimental) to use `babel-macros` to do the same 🎉 

I can invest some time in getting this all hooked up, but not sure what's the priority of this, if the ROI is big enough, I can probably pull this off in a day. @nkohari, need your help here!

Related: `color` should probably come from our tokens instead of `Proptypes.string`? @landitus, wdyt?

**Comment by landitus** [comment]
My take: we could do it after the PoC.
- I don't mind having a string for icons, at least for now. Having a dropdown of icons is a good enhancement to the docs.
- Just to take into account: Should we need support for custom icons in the future; maybe not passing a name and passing a svg directly?
- With the colors, yes: having support for choosing colors from tokens sound good. We would need a dropdown with images so that it would support having a swatch of color and the name of the token. This enhancement would also fit the icon dropdown as well.

**Comment by siddharthkp** [comment]
Fixed in https://github.com/auth0/cosmos/pull/546
